element/src/commonMain/kotlin/RoundedRectangleElement.kt

```diff
@@ -40,3 +46,21 @@ public class RoundedRectangleElement : Element() {
         override fun trySelect(element: Element): RoundedRectangleElement? = element as? RoundedRectangleElement
     }
 }
+
+/** Transforms a radius such that requested radii which are larger than their rectangle behave nicely. */
+private fun safeRadius(desired: Float, horizontalNeighbor: Float, verticalNeighbor: Float, width: Float, height: Float): Float {
+    val safeHorizontal = (desired + horizontalNeighbor) <= width
+    val safeVertical = (desired + verticalNeighbor) <= height
+    return if (safeHorizontal && safeVertical) {
+        desired
+    } else if (!safeHorizontal && !safeVertical) {
+        minOf(
+            width * desired / (desired + horizontalNeighbor),
+            height * desired / (desired + verticalNeighbor)
+        )
+    } else if (!safeHorizontal) {
+        width * desired / (desired + horizontalNeighbor)
+    } else { // !safeVertical
+        height * desired / (desired + verticalNeighbor)
+    }
```

Comment:
nit: inverse booleans are slightly harder for me to read, might be easier to read as?:

```suggestion
    return if (safeHorizontal && safeVertical) {
        desired
    } else if (safeHorizontal) {
        height * desired / (desired + verticalNeighbor)
    } else if (safeVertical) {
        width * desired / (desired + horizontalNeighbor)
    } else { // !safeHorizontal && !safeVertical
        minOf(
            width * desired / (desired + horizontalNeighbor),
            height * desired / (desired + verticalNeighbor)
        )
    }
```

_I think I translated that correctly._ 🤞